samples/snippets/src/main/java/com/example/spanner/PgInterleavedTableSample.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner;
+
+// [START spanner_postgresql_interleaved_table]
+
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerExceptionFactory;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.spanner.admin.database.v1.UpdateDatabaseDdlMetadata;
+import java.util.Arrays;
+import java.util.concurrent.ExecutionException;
+
+public class PgInterleavedTableSample {
+
+  static void pgInterleavedTable() {
+    // TODO(developer): Replace these variables before running the sample.
+    final String projectId = "my-project";
+    final String instanceId = "my-instance";
+    final String databaseId = "my-database";
+    pgInterleavedTable(projectId, instanceId, databaseId);
+  }
+
+  static void pgInterleavedTable(String projectId, String instanceId, String databaseId) {
+    try (Spanner spanner =
+        SpannerOptions.newBuilder()
+            .setProjectId(projectId)
+            .build()
+            .getService()) {
+      final DatabaseAdminClient databaseAdminClient = spanner.getDatabaseAdminClient();
+
+      // The Spanner PostgreSQL dialect extends the PostgreSQL dialect with certain Spanner
+      // specific features, such as interleaved tables.
+      // See https://cloud.google.com/spanner/docs/postgresql/data-definition-language#create_table
+      // for the full CREATE TABLE syntax.
+      final OperationFuture<Void, UpdateDatabaseDdlMetadata> updateOperation =
+          databaseAdminClient.updateDatabaseDdl(
+              instanceId,
+              databaseId,
+              Arrays.asList(
+                  "CREATE TABLE Singers ("
+                      + "  SingerId  bigint NOT NULL PRIMARY KEY,"
+                      + "  FirstName varchar(1024) NOT NULL,"
+                      + "  LastName  varchar(1024) NOT NULL"
+                      + ")",
+                  "CREATE TABLE Albums ("
+                      + "  SingerId bigint NOT NULL,"
+                      + "  AlbumId  bigint NOT NULL,"
+                      + "  Title    varchar(1024) NOT NULL,"
+                      + "  PRIMARY KEY (SingerId, AlbumId)"
+                      + ") INTERLEAVE IN PARENT Singers ON DELETE CASCADE"),
```

Comment:
Do we have sample for drop?  One of the benefit is that we only support cascade drop for interleaved table.

Reply:
What exactly do you mean? I don't think we support cascade drop, only cascade delete. Or is that what you meant?